src/main/java/core/basesyntax/service/impl/ParserImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.Parser;
+
+public class ParserImpl implements Parser {
+    private static final String DELIMITER = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public FruitTransaction parse(String data) {
+        if (!data.matches("^[brsp],.*")) {
```

Comment:
regex validation is an interesting idea, let's do it fully, and extract PATTERN to a constant
```suggestion
        if (!data.matches("^[brsp],[a-zA-Z]+,[0-9]+$")) {
```